### ISSUE TITLE
Inconsistent file attachment renaming

### ISSUE THREAD

**Issue by fabianoriccardi** [laurent22/joplin#11237]
### Operating system

Windows

### Joplin version

3.0.15

### Desktop version info

Joplin 3.0.15 (prod, win32)

Client ID: ab5b95c425084ccb936f12ed2f6e76d1
Sync Version: 3
Profile Version: 47
Keychain Supported: Yes

Revision: 598677b

Backup: 1.4.1

### Current behaviour

I can apparently rename an attachment by editing the markdown:

    [the_file.txt](:/1274a0bedcc04656878e2f0dd0014114)

But if I try to "save as" by right clicking from the non-markdown editor, the location picker shows the original file name (i.e. the name that the file had when I attached it to the note). 

### Expected behaviour

Instead, I would see as predefined what I have written in markdown (e.g. *the_file.txt*).

**Comment by dpoulton-github** [comment]
> I can apparently rename an attachment by editing the markdown:

I do not think that is correct. The text in the square brackets is not the filename it is the *link text*. This allows a hyperlink to be made with context, just like on a webpage.

Example:

I attach a file called "2023-12-01_export_system_status-4a2dec32.pdf" and Joplin stores this as `1274a0bedcc04656878e2f0dd0014114`. When writing a note I can change the *text of the hyperlink*  so that it makes contextual sense.

`The [previous system status report](:/1274a0bedcc04656878e2f0dd0014114) did not show anything unusual`

But if I right-click and "save as" it saves the file as "2023-12-01_export_system_status-4a2dec32.pdf".

**Comment by fabianoriccardi** [comment]
I had written **apparently** because I know this should not be intended as file renaming function. 

However, I would really have the capability to rename an attachment, independently from the text in square brackets, so that if I try to "save as", the proposed name reflects the new file name.

Probably this is not a bug, but a feature request...
